### ISSUE TITLE
[RESOLVED] ⚠️Production outage (partial): bot ceased receiving events for certain servers from Discord after about 2022-12-14 11:01 UTC

### ISSUE THREAD

**Issue by kevinlul** [DawnbrandBots/bastion-bot#246]
Resolved 2022-12-14T12:27:11.230Z by container reboot.

Certain shards disconnected from the Discord gateway and either failed to reconnect, or Discord ceased sending events for those shards to the bot. This resulted in a partial outage for about ~1.5 hours until operator intervention. Some servers would be affected and others would not be, depending on the shard they were assigned. Shard 0 is the only one that was certainly not affected by this, which includes direct messages.

Log excerpt:

```
2022-12-14T11:00:06.776Z bot:info:abdeploy Updating from /var/local/bastion/abdeploy.json
2022-12-14T11:00:06.777Z bot:info:abdeploy Read 2 entries
USER SEARCHES
2022-12-14T11:01:00.760Z bot:info:events Shard 1 reconnecting
2022-12-14T11:01:00.945Z bot:info:events Shard 1 resumed: 1 events replayed
2022-12-14T11:01:05.761Z bot:info:events Shard 1 reconnecting
2022-12-14T11:01:25.709Z bot:info:events Shard 5 reconnecting
2022-12-14T11:01:25.971Z bot:info:events Shard 5 resumed: 38 events replayed
2022-12-14T11:01:30.711Z bot:info:events Shard 5 reconnecting
2022-12-14T11:02:18.611Z bot:info:events Shard 4 reconnecting
2022-12-14T11:02:22.038Z bot:info:events Shard 6 reconnecting
2022-12-14T11:02:24.778Z bot:notify:events Shard 4 ready
2022-12-14T11:02:27.827Z bot:notify:events Shard 6 ready
2022-12-14T11:02:43.612Z bot:info:events Shard 4 reconnecting
2022-12-14T11:02:47.039Z bot:info:events Shard 6 reconnecting
2022-12-14T11:03:05.482Z bot:info:events Shard 2 reconnecting
USER SEARCHES
2022-12-14T12:26:21.141Z OPERATOR ATTEMPTS CONTAINER REBOOT
```

**Comment by kevinlul** [comment]
Possible root cause for all five incidents: discordjs/discord.js#8486

Past incidents: #199, #204, #212, #233

**Comment by kevinlul** [comment]
What a patch https://github.com/NeonWizard/sockbot-discord/blob/main/src/modules/wspatch.ts

**Comment by kevinlul** [comment]
Another incident around 2023/03/08 15:57 ET

**Comment by kevinlul** [comment]
Fixed in a07956876827cd6d3ae06e94313e6bc29f799931 (discordjs/discord.js#8989 is included in 14.8.0 and closed discordjs/discord.js#8486) unless proven otherwise
